packages/edit-site/src/store/utils.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * External dependencies
+ */
+import memoize from 'memize';
+
+/**
+ * WordPress dependencies
+ */
+import { isTemplatePart } from '@wordpress/blocks';
+
+const EMPTY_ARRAY = [];
+
+/**
+ * Get a flattened and filtered list of template parts and the matching block for that template part.
+ *
+ * Takes a list of blocks defined within a template, and a list of template parts, and returns a
+ * flattened list of template parts and the matching block for that template part.
+ *
+ * @param {Array}  blocks        Blocks to flatten.
+ * @param {?Array} templateParts Available template parts.
+ * @return {Array} An array of template parts and their blocks.
+ */
+export function getFilteredTemplatePartBlocks(
```

Comment:
I think it can be confusing if we export both the non-memoized and the memoized versions. 

Could we just export `memoizedGetFilteredTemplatePartBlocks` and ideally, export it as `getFilteredTemplatePartBlocks` to avoid that confusion?

I see we're exposing the non-memoized function for unit testing purposes, but it should be fine to test the memoized function and just add a `memoizedGetFilteredTemplatePartBlocks.clear()` before or after every test, right?

Reply:
Looks like it works pretty well exporting the memoized version as `getFilteredTemplatePartBlocks`, and neatens things up a fair bit. Good idea 👍